rotkehlchen/data_migrations/migrations/migration_11.py

```diff
@@ -42,10 +41,49 @@ def _get_optimism_transaction_fees(rotki: 'Rotkehlchen', progress_handler: 'Migr
         )
 
 
+def _get_optimism_transaction_fees_num_steps(rotki: 'Rotkehlchen') -> int:
+    """Returns the number of progress handler steps needed for the get optimism transaction fees"""
+    with rotki.data.db.conn.read_ctx() as cursor:
+        cursor.execute('SELECT tx_hash, identifier FROM evm_transactions WHERE chain_id=10')
+        txhash_to_id = dict(cursor)
+    return len(txhash_to_id)
+
+
+def _detect_arbitrum_one_accounts_num_steps(rotki: 'Rotkehlchen') -> int:
```

Comment:
no need to make a function for every little thing. Just calculate both step parts under the same DB cursor please and be done with it. Don't make a function for each little subthing that you are only ever gonna use once.

Also this is wrong. The function you used calculate's detects addresses that have had activity for all supported EVM chains.